src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -244,98 +235,107 @@ private void verifyAttribute(AttributedElement ae, Attribute<?> a, List<VerifyEr
                 }
-                size = 2 + 2 * nma.nestMembers().size();
+                yield 2 + 2 * nma.nestMembers().size();
             }
             case PermittedSubclassesAttribute psa -> {
                 if (classModel.flags().has(AccessFlag.FINAL)) {
                     errors.add(new VerifyError("PermittedSubclasses attribute in final %s".formatted(toString(ae))));
                 }
-                size = 2 + 2 * psa.permittedSubclasses().size();
-            }
-            case RecordAttribute ra -> {
-                size = 2;
-                for (var rc : ra.components()) {
-                    size += 6;
-                    for (var rca : rc.attributes()) {
-                        size += 6 + ((BoundAttribute)rca).payloadLen();
-                    }
-                }
+                yield 2 + 2 * psa.permittedSubclasses().size();
             }
+            case RecordAttribute ra ->
+                componentsSize(ra.components());
             case RuntimeVisibleAnnotationsAttribute aa ->
-                size = annotationsSize(aa.annotations());
+                annotationsSize(aa.annotations());
             case RuntimeInvisibleAnnotationsAttribute aa ->
-                size = annotationsSize(aa.annotations());
-            case RuntimeVisibleParameterAnnotationsAttribute aa -> {
-                size = 1;
-                for (var ans : aa.parameterAnnotations()) {
-                    size += annotationsSize(ans);
-                }
-            }
-            case RuntimeInvisibleParameterAnnotationsAttribute aa -> {
-                size = 1;
-                for (var ans : aa.parameterAnnotations()) {
-                    size += annotationsSize(ans);
-                }
-            }
+                annotationsSize(aa.annotations());
+            case RuntimeVisibleParameterAnnotationsAttribute aa ->
+                parameterAnnotationsSize(aa.parameterAnnotations());
+            case RuntimeInvisibleParameterAnnotationsAttribute aa ->
+                parameterAnnotationsSize(aa.parameterAnnotations());
             case SignatureAttribute sa -> {
                 sa.signature();
-                size = 2;
-            }
-            case SourceDebugExtensionAttribute sda -> {
-                size = sda.contents().length;
+                yield 2;
             }
+            case SourceDebugExtensionAttribute sda ->
+                sda.contents().length;
             case SourceFileAttribute sfa -> {
                 sfa.sourceFile();
-                size = 2;
+                yield 2;
             }
             case SourceIDAttribute sida -> {
                 sida.sourceId();
-                size = 2;
+                yield 2;
             }
-            case SyntheticAttribute sa ->
-                size = 0;
-            default -> {}
-        }
+            case SyntheticAttribute _ ->
+                0;
+            default ->
+                -1;
```

Comment:
I recommend we explicitly return -1 to skip verification only for UnknownAttribute and CustomAttribute; then our tests can catch missing verification for new attribute additions.

Reply:
There are still many attributes missing here (yet). However it is a nice goal.